sdk/cosmos/azure-cosmos-encryption/src/main/java/com/azure/cosmos/encryption/EncryptionCosmosAsyncContainer.java

```diff
@@ -33,13 +33,16 @@
 
 // TODO: for now basic functionality is in. some APIs and some logic branch is not complete yet.
 public class EncryptionCosmosAsyncContainer {
+    private final Scheduler encryptionScheduler;
     private final Encryptor encryptor;
     private final CosmosResponseFactory responseFactory = new CosmosResponseFactoryCore();
     private final CosmosAsyncContainer container;
 
     EncryptionCosmosAsyncContainer(String id, CosmosAsyncDatabase database, Encryptor encryptor) {
         this.container = BridgeInternal.createCosmosAsyncContainer(id, database);
         this.encryptor = encryptor;
+        // TODO: moderakh once EncryptionKeyWrapProvider apis are made async this should change to Schedulers.parallel()
+        this.encryptionScheduler = Schedulers.elastic();
```

Comment:
By capturing the `Schedulers.elastic()` in an instance variable, you are using the same Scheduler with every async API. Just a heads up on not sure how this will behave. 

Also, `Schedulers.boundedElastic()` is the preferred way now for wrapping I/O calls. Please see that too.

Reply:
using the variable is fine. The elastic scheduler provided by the reactor lib is a singleton instance.

sure, I will switch to `Schedulers.boundedElastic()` for now (next commit). in the long run this will be replaced by `Schedulers.parallel()` after I fix the remaining blocking io call. (one potential blocking call is remaining).